File: webhole/packages/php/cat.php
<?php
function cat($path, $old_path = null) {
    if ($old_path !== null) {
        chdir($old_path);
    }

    // Check if the file exists and is a regular file
    if (!is_file($path)) {
        return "Error: '$path' is not a valid file.";
    }

    // Attempt to read the file
    try {
        return file_get_contents($path);
    } catch (Exception $e) {
        return "Error reading file: " . $e->getMessage();
    }
}

echo cat("__path__");

File: webhole/packages/python-django/cat.py
def cat(path, old_path=None):
    import os
    os.chdir(old_path)
    """Reads and prints the content of a file."""
    if not os.path.isfile(path):
        return f"Error: '{path}' is not a valid file."
    try:
        with open(path, 'r', encoding='utf-8') as file:
            return file.read()
    except Exception as e:
        return f"Error reading file: {e}"

print(cat("__path__"))